clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -50,7 +50,7 @@
 
     // This matches URIs of formats: host:port and protocol:\\host:port
     // IPv6 is supported with [ip] pattern
-    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?\\[?([0-9a-zA-Z\\-.:]*)\\]?:([0-9]+)");
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile(".*?(?:(?:\\[([0-9a-zA-Z:]*)\\])|([0-9a-zA-Z\\-.]*))(?::([0-9]+))?");
```

Comment:
.\*?  **<- pattern to match the protocol part of the url**
    (?:(?:\\[([0-9a-zA-Z:]\*)\\])  **<- match the host as an IPv6 address**
              |([0-9a-zA-Z\\-.]\*))  **<- match the host as a non-IPv6 address (doesn't contain ':')**
(?::([0-9]+))? **<-match the "optional" port number**